contracts/consensus-gateway/ConsensusCogateway.sol

```diff
@@ -85,4 +98,81 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
             _maxStorageRootItems
         );
     }
+
+    /**
+     * @notice This method will be called by anyone to verify merkle proof of
+     *          gateway contract address.
+     *
+     *  @param _blockHeight Block height at which Gateway is to be proven.
+     *  @param _rlpAccount RLP encoded account node object.
+     *  @param _rlpParentNodes RLP encoded value of account proof node array.
+     */
+    function proveConsensusGateway(
+        uint256 _blockHeight,
+        bytes calldata _rlpAccount,
+        bytes calldata _rlpParentNodes
+    )
+        external
+    {
+        MessageInbox.proveStorageAccount(
+            _blockHeight,
+            _rlpAccount,
+            _rlpParentNodes
+        );
+    }
+
+    /**
+     * @notice Confirms the initiation of opening a kernel at auxiliary chain.
+     *
+     * @param _kernelHeight Height of the kernel.
+     * @param _kernelHash Hash of the kernel.
+     * @param _feeGasPrice Gas price which the sender is willing to pay.
+     * @param _feeGasLimit Gas limit which the sender is willing to pay.
+     * @param _sender Sender address.
+     * @param _blockHeight Block height for which the proof is valid.
+     * @param _rlpParentNodes RLP encoded parent node data to prove message
+     *                        exists in outbox of ConsensusGateway.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmOpenKernel(
+        uint256 _kernelHeight,
+        bytes32 _kernelHash,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _sender,
```

Comment:
Validation that sender address cannot be address(0)?

Reply:
Done